### PR TITLE
Simplify how we detect fips in chef-config

### PR DESCRIPTION
**Pull request by tas50**
Use the same code we use in ohai.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: chef-config/lib/chef-config/fips.rb [hunk 1]
```diff
@@ -21,33 +21,11 @@
...
-        false
-      end
+    require "openssl" unless defined?(OpenSSL)
+    if defined?(OpenSSL.fips_mode) && OpenSSL.fips_mode
+      true
     else
-      fips_path = "/proc/sys/crypto/fips_enabled"
-      File.exist?(fips_path) && File.read(fips_path).chomp != "0"
+      false
     end
```
Comment (lamont-granquist) [inline chef-config/lib/chef-config/fips.rb:29]: kill the if-else and just return the boolean expression

Comment (lamont-granquist) [reply chef-config/lib/chef-config/fips.rb:29]: (there should be a rubocop rule for this if there isn't one already

Comment (lamont-granquist) [reply chef-config/lib/chef-config/fips.rb:29]: someone needs to write a rule to finding unmatched parens in comments as well)

Comment (lamont-granquist) [reply chef-config/lib/chef-config/fips.rb:29]: (although someone will ban multiple nesting levels of parents (and i'd hate that))

Comment (lamont-granquist) [reply chef-config/lib/chef-config/fips.rb:29]: *parens

Comment (tas50) [reply chef-config/lib/chef-config/fips.rb:29]: Fixed. I wonder how that ended up that way in Ohai. There must have been more there to start.

**Comment by tas50** [comment]
In order to go down this route we're going to need to do some fips testing on real boxes. Ideally this is 1:1 compatible, but faster than the old method, but who knows in FIPS-land.

File: chef-config/lib/chef-config/fips.rb (hunk 1 above)
Comment (phiggins) [reply chef-config/lib/chef-config/fips.rb]: > there should be a rubocop rule for this if there isn't one already

My continual rubocop experience is being surprised it doesn't handle all of my annoyances.

**Pull request closed without merging**